### PR TITLE
new raw blocks: listen, frontend, backend

### PR DESCRIPTION
**Pull request by kinjelom**
Proper formatting of indented multiline configurations, new raw blocks: listen, frontend, backend.

### CODE REVIEW

File: jobs/haproxy/templates/haproxy.config.erb [hunk 1]
```diff
@@ -1,5 +1,21 @@
 <%- # see https://bosh.io/docs/jobs/#properties for documentation of bosh ERB templates -%>
-<% if properties.ha_proxy.raw_config -%>
+<%
+  def format_indented_multiline_config(raw_config)
```
Comment (maxmoehl) [inline jobs/haproxy/templates/haproxy.config.erb:3]: This was bugging me for quite some time, thanks for taking the time to fix this 😄

**Review by maxmoehl — DISMISSED** [review]
Thank you for your contribution!

Can you please [add some tests](https://github.com/cloudfoundry/haproxy-boshrelease/tree/master/spec/haproxy/templates)? Then we can run it through our CI to see if everything works as expected.

File: jobs/haproxy/spec [hunk 2]
```diff
@@ -671,6 +671,36 @@ properties:
...
+    description: |
+      A hash of 'backend' blocks with the configuration as a multiline text blob or an array of lines (for better diffs).
+      Use at your own risk.
+    example:
+      my-backend-x: |
+        mode http
+        server-template srv-x 1-3 q-s0.web.default.deployment-x.bosh:8080 check inter 1000
+      my-backend-y: |
+        - mode http
+        - server-template srv-y 1-3 q-s0.web.default.deployment-y.bosh:8080 check inter 1000
```
[the remark this answers is not in the document]
Comment (kinjelom) [reply jobs/haproxy/spec:702]: Yes, this is just an example. By mistake, I combined multiline text and an array.

**Review by b1tamara — APPROVED** [review]
LGTM
Unit-tests and acceptance-test are green

**Review by b1tamara — APPROVED** [review]
LGTM

**Comment by kinjelom** [comment]
thank you :+1:

**Pull request merged**